modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClayButton from '../shared/ClayButton.es';
+import getCN from 'classnames';
+
+class Conjunction extends React.Component {
+	_getConjunctionLabel(conjunctionName, conjunctions) {
+		const conjunction = conjunctions.find(
+			({name}) => name === conjunctionName
+		);
+
+		return conjunction ? conjunction.label : undefined;
+	}
+	render() {
+		const {
+			className,
+			conjunctionName,
+			editing,
+			supportedConjunctions
+		} = this.props;
+
+		const classnames = getCN(
+			{
+				'btn-sm conjunction-button': editing,
+				'conjunction-label': !editing
+			},
+			className
+		);
+
+		return (
+			editing ?
+				<ClayButton
+					className={classnames}
+					label={this._getConjunctionLabel(
+						conjunctionName,
+						supportedConjunctions
+					)}
+					onClick={this.props.handleConjunctionClick}
```

Comment:
Internal methods should use the keyword `handle` but props should be `on`. So this should be `onConjunctionClick`. The reason for this is to follow JSX pattern of `onClick`.

Reply:
Also this can be destructured from the other props